### PR TITLE
ROX-11166: composite data for indirectly scoped resources

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

This change aims at generating pseudo-realistic dackbox datasets for SAC integration testing.
Generating realistic datasets is time consuming and not necessarily helpful when it comes to SAC testing, as it in not easy to visualize how much data is present in the storage, nor how the items are related between themselves.

Having a simple, low-cardinality data set allow to more easily focus on what items should be reachable and which not when it comes to scoped access control and scoped data filtering testing.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

This change aims at creating a dataset for testing purposes. Actual tests will come in a later pull request.
CI run should be enough.

### CODE REVIEW

File: pkg/fixtures/dackbox.go [hunk 1]
```diff
@@ -0,0 +1,1128 @@
...
+//                                                               |          +--> ImageCVE5    |
+//                                                   +-----------+                            |
+//                                                   |                                        |
+//                                                   +--> ImageComponent4                     |
+//                                                   |                                        |
+//                                                   |                      +-----------------+
+//                                                   |                      |
+// Cluster2 -- NamespaceB -- Deployment2 -- Image2 --+--> ImageComponent5 --+--> ImageCVE6
+//                                                                          |
+//                                                                          +--> ImageCVE7
```
Comment (janisz) [inline pkg/fixtures/dackbox.go:86]: ❤️

File: pkg/fixtures/dackbox.go [hunk 2]
```diff
@@ -0,0 +1,1128 @@
...
+// ImageComponentEdge         |
+//         |                  |
+//         v *                |
+//   ImageComponent      ImageCVEEdge
+//         ^ *                |
+//         |                  |
+//  ComponentCVEEdge          |
+//         |                  |
+//         v *                |
+//        CVE <---------------+ *
```
Comment (janisz) [inline pkg/fixtures/dackbox.go:37]: Maybe we should have a file with fill schema diagram generated from source? What do you think about it?

File: pkg/fixtures/dackbox.go [hunk 3]
```diff
@@ -0,0 +1,1162 @@
...
+)
+
+///////////////////////////////
+// Dackbox synthetic dataset //
+///////////////////////////////
+
+/////////////////////
+// Image with CVEs //
+
+// Data relationships
```
Comment (dhaus67) [inline pkg/fixtures/dackbox.go:17]: not related at all: appreciate the diagram, helps better in understanding things 😸

File: pkg/fixtures/dackbox.go [hunk 4]
```diff
@@ -0,0 +1,1162 @@
...
+			Score:               5.9,
+			Severity:            storage.CVSSV3_MEDIUM,
+		},
+		PublishedOn:       &types.Timestamp{Seconds: 1234567890},
+		LastModified:      &types.Timestamp{Seconds: 1235467890},
+		VulnerabilityType: storage.EmbeddedVulnerability_IMAGE_VULNERABILITY,
+		VulnerabilityTypes: []storage.EmbeddedVulnerability_VulnerabilityType{
+			storage.EmbeddedVulnerability_IMAGE_VULNERABILITY,
+		},
+		Suppressed:            false,
```
Comment (dhaus67) [inline pkg/fixtures/dackbox.go:437]: <strike>Would it be worthwhile to add suppressed CVE's as well?</strike>

I see we haven't implemented that yet.

File: pkg/fixtures/dackbox.go [hunk 5]
```diff
@@ -0,0 +1,1162 @@
...
+				Id:   "169b0d3f-8277-4900-bbce-1127077defae",
+				Name: "Stackrox Scanner",
+			},
+			Notes: []storage.ImageScan_Note{},
+		},
+		SignatureVerificationData: nil,
+		Signature:                 nil,
+		SetComponents:             &storage.Image_Components{Components: 3},
+		SetCves:                   &storage.Image_Cves{Cves: 5},
+		SetFixable:                &storage.Image_FixableCves{FixableCves: 2},
```
Comment (dhaus67) [inline pkg/fixtures/dackbox.go:629]: Not sure I'm reading things right, but shouldn't it be logically 5 fixable CVE's?

All three embedded image components have the `FixedBy` field set, all three components are different and `1_1 & 1s2_3` have each respectively 2 CVE's and `1_2` has 1 CVE, or are my assumptions wrong that they are not somehow fixed?

File: pkg/fixtures/dackbox.go [hunk 6]
```diff
@@ -0,0 +1,1162 @@
...
+			Components: []*storage.EmbeddedNodeScanComponent{
+				GetEmbeddedNodeComponent_1_1(),
+				GetEmbeddedNodeComponent_1_2(),
+				GetEmbeddedNodeComponent_1s2_3(),
+			},
+			Notes: nil,
+		},
+		SetComponents: &storage.Node_Components{Components: 3},
+		SetCves:       &storage.Node_Cves{Cves: 5},
+		SetFixable:    &storage.Node_FixableCves{FixableCves: 2},
```
Comment (dhaus67) [inline pkg/fixtures/dackbox.go:1102]: I guess same question here as above: in the current mock data, how is a node CVE classified as fixable or not fixable? I'd expect the `storage.EmbeddedVulnerability.SetFixedBy` to be it?

File: pkg/fixtures/dackbox.go (hunk 4 above)
Comment (rhybrillou) [reply pkg/fixtures/dackbox.go:437]: Good point.
I think I'd use the datastore Suppress/Unsuppress functions to test the behaviour, though.

File: pkg/fixtures/dackbox.go (hunk 5 above)
Comment (rhybrillou) [reply pkg/fixtures/dackbox.go:629]: To be honest, I did not dig that deep in the effort to provide pseudo-realistic test data.
My guess would be that the `SetFixedBy` field in the CVEs from the graph is what is actually used to consider a CVE as fixed. I think the `FixableCves` value is the count of visible CVEs that are flagged as fixable.
Although the values are not relevant in the tests I intend to run, I adjusted the dataset to be consistent.

**Review by dhaus67 — APPROVED** [review]

**Pull request merged**